Framework/Algorithms/test/UnGroupWorkspaceTest.h

```diff
@@ -125,4 +148,16 @@ class UnGroupWorkspaceTest : public CxxTest::TestSuite {
         ads.remove(member);
     }
   }
+
+  bool isAlgorithmInHistory(const Mantid::API::Workspace_sptr ws) {
+    auto ws_histories = ws->getHistory().getAlgorithmHistories();
+    bool found = false;
+    for (auto alg_history : ws_histories) {
+      if (alg_history->name() == "UnGroupWorkspace") {
+        found = true;
+        break;
+      }
+    }
+    return found;
```

Comment:
I think we can use a c++ std library algorithm here
```suggestion
    auto iter = std::find_if (wsHistories.cbegin(), wsHistories.cend(), [](auto const &algHistory){ return algHistory->name() == "UnGroupWorkspace";});
    return iter != wsHistories.cend();
```